### DISCUSSION TITLE
error: Compiler does not support fentry

### DISCUSSION THREAD

**Discussion by karim-manaouil** [bpftrace/bpftrace#3030] [Q&A] [answered]
Hi,

I tried to include `<linux/perf_event.h>`  to have access to `struct pmu`  in the following program 

```
#include <linux/perf_event.h>

k:perf_pmu_enable {
        $s = ksym(((struct pmu *)arg0)->pmu_enable);
        printf("%s\n", $s);
}

```

and got the following error coming from `ftrace.h`.

```
/lib/modules/6.7.6+/build/arch/x86/include/asm/processor.h:519:17: warning: taking address of packed member 'sp0' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/6.7.6+/build/arch/x86/include/asm/ftrace.h:7:3: error: Compiler does not support fentry?
/lib/modules/6.7.6+/build/include/linux/cgroup-defs.h:556:16: warning: field 'cgrp' with variable sized type 'struct cgroup' not at the end of a struct or class is a GNU extension [-Wgnu-variable-sized-type-not-at-end]
```

Probably because clang doesn't support fentry. Is there anything I can do about it?

Edit: added bpftrace program

**Comment by danobi** [comment]
Hi, can you please share the script you tried to run?

**Reply by karim-manaouil** [reply]
@danobi I edited the post and added the script

**Reply by danobi** [reply]
Thanks!

**Comment by danobi** [comment, marked as answer]
The script fails for me as well. But if I remove the `#include`, it works. Can you try that? 

See https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#btf-support for more details about why removing the `#include` helps

**Reply by karim-manaouil** [reply]
Works! Awesome! Thanks for pointing out the BTF stuff! Yaay!! No more `#includes`!
